packages/aws-cdk-lib/aws-logs/lib/log-retention-provider/index.ts

```diff
@@ -121,6 +121,64 @@ async function setRetentionPolicy(logGroupName: string, region?: string, options
   } while (true); // exit happens on retry count check
 }
 
+/**
+ * Tags and untags a log group. This includes adding new tags and updating existing tags.
+ *
+ * @param logGroupName the name of the log group to create
+ * @param tags the tags to propagate to the log group
+ * @param region the region of the log group
+ * @param options CloudWatch API SDK options
+ */
+async function setLogGroupTags(logGroupName: string, tags: AWS.CloudWatchLogs.Tags[], region?: string, options?: SdkRetryOptions) {
+  // The same as in createLogGroupSafe(), here we could end up with the race
+  // condition where a log group is either already being created or its retention
+  // policy is being updated. This would result in an OperationAbortedException,
+  // which we will try to catch and retry the command a number of times before failing
+  let retryCount = options?.maxRetries == undefined ? 10 : options.maxRetries;
+  const delay = options?.retryOptions?.base == undefined ? 10 : options.retryOptions.base;
+  do {
+    try {
+      const cloudwatchlogs = new AWS.CloudWatchLogs({ apiVersion: '2014-03-28', region, ...options });
+      const tagsOnLogGroup = (await cloudwatchlogs.listTagsLogGroup({ logGroupName }).promise()).tags ?? {};
+
+      const tagsToSet: { [key: string]: string } = {};
+      const tagsKeys: string[] = [];
+      for (const tag of tags) {
+        if (tagsOnLogGroup[tag.Key] === undefined || tagsOnLogGroup[tag.Key] !== tag.Value) {
+          tagsToSet[tag.Key] = tag.Value;
+        }
+        tagsKeys.push(tag.Key);
+      }
+
+      const tagsToDelete = tagsOnLogGroup
+        ? Object.keys(tagsOnLogGroup).filter(tag => !tagsKeys.includes(tag))
+        : [];
+
+      if (Object.keys(tagsToSet).length > 0) {
+        await cloudwatchlogs.tagLogGroup({ logGroupName, tags: tagsToSet }).promise();
```

Comment:
It looks like you can associate as many as 50 tags with a CloudWatch Logs
resource. We should handle this somehow. We could probably start off by just
throwing a validation error in the construct if > 50. I'm not sure how likely
that is so if we receive issue then we could handle > 50 at that point (I would
guess it's not that likely)

Reply:
I've updated this to use tagResource and have updated the condition to check for a length greater than 0 and less than or equal to 50.